packages/react-reconciler/src/ReactChildFiber.js

```diff
@@ -267,12 +260,10 @@ function coerceRef(
   let coercedRef;
   if (
     !disableStringRefs &&
-    mixedRef !== null &&
-    typeof mixedRef !== 'function' &&
-    typeof mixedRef !== 'object'
+    (typeof mixedRef === 'string' ||
+      typeof mixedRef === 'number' ||
+      typeof mixedRef === 'boolean')
```

Comment:
This is a subtle change because we used to coerce any non-function, non-object type to a string, but now we explicitly check for only these types. It's only when string refs are not disabled, though.